wpa_supplicant/events.c

```diff
@@ -2947,8 +2947,15 @@ static void wnm_process_assoc_resp(struct wpa_supplicant *wpa_s,
 #ifdef CONFIG_SME
 	if (elems.bss_max_idle_period) {
 		unsigned int msec;
-		wpa_s->sme.bss_max_idle_period =
-			WPA_GET_LE16(elems.bss_max_idle_period);
+
+		if (wpa_s->conf->max_idle_period >=
+			WPA_GET_LE16(elems.bss_max_idle_period)) {
+			wpa_s->sme.bss_max_idle_period =
+				WPA_GET_LE16(elems.bss_max_idle_period);
+		} else {
+			wpa_s->sme.bss_max_idle_period = wpa_s->conf->max_idle_period;
+		}
+
```

Comment:
as per the spec, the value in assoc request is just a preference, we should always use the value from the assoc response as is: See `802.11-2024: 11.21.13`

> If the BSS Max Idle Period element is present in the (Re)Association Request frame received by a non-S1G AP that has dot11BSSMaxIdlePeriodIndicationByNonAPSTA equal to true, then the non-S1G AP may choose the non-AP STA’s preferred maximum idle period. The non-S1G AP indicates its chosen value to the non-S1G STA in the (Re)Association Response frame.